RevenueCatUI/CustomerCenter/ViewModels/PromotionalOfferViewModel.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PromotionalOfferViewModel.swift
+//
+//
+//  Created by Cesar de la Vega on 17/6/24.
+//
+
+import Foundation
+import RevenueCat
+
+#if os(iOS)
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@available(visionOS, unavailable)
+@MainActor
+class PromotionalOfferViewModel: ObservableObject {
+
+    @Published
+    private(set) var promotionalOfferData: PromotionalOfferData?
+    @Published
+    private(set) var error: Error?
+
+    private var purchasesProvider: CustomerCenterPurchasesType
+    private let loadPromotionalOfferUseCase: LoadPromotionalOfferUseCase
+
+    convenience init() {
+        self.init(promotionalOfferData: nil)
+    }
+
+    init(promotionalOfferData: PromotionalOfferData?) {
+        self.promotionalOfferData = promotionalOfferData
+        self.purchasesProvider = CustomerCenterPurchases()
+        self.loadPromotionalOfferUseCase = LoadPromotionalOfferUseCase()
+    }
+
+    func purchasePromo() async {
+        guard let promotionalOffer = self.promotionalOfferData?.promotionalOffer,
+              let product = self.promotionalOfferData?.product else {
+            Logger.warning(Strings.promo_offer_not_loaded)
+            return
+        }
+
+        do {
+            let result = try await Purchases.shared.purchase(product: product, promotionalOffer: promotionalOffer)
```

Comment:
Thinking about this, we should move this to the `CustomerCenterPurchasesType`?

Reply:
we should for sue. I will do it in another PR if you don't mind, since that's going to involve some updates and potentially breaking tests and I need to merge this today